### PR TITLE
fix(common): live-9252 update buy/sell api and expose methods in onRampProvider

### PR DESCRIPTION
**Pull request by beths-ledger**
### 📝 Description

The onRampProvider was using out of date cached CDN endpoint, which hid available currencies like NEAR. 
This PR updates the API that informs whether currencies can be bought or sold so that new currencies like NEAR show as available.
- Expose `isCurrencySupported` and `getSupportedCurrencyIds` methods from `useRampProvider` hook.
- Add unit tests for helper functions

Updates screenshots to reflect **only** the new  **correct** API responses:

1. "binance_beacon_chain". i.e. BNB Should NOT be available to BUY ✅
2. "cardano" SHOULD be available to BUY ✅
3. "solana" SHOULD be available to SWAP and BUY ✅
4. "ethereum/erc20/usd__coin"  i.e. USDC  SHOULD be available to SWAP ✅

### ❓ Context

- **Impacted projects**: `ledger-live-common` `ledger-live-desktop` `ledger-live-mobile` - 
- **Linked resource(s)**:  https://ledgerhq.atlassian.net/browse/LIVE-9252?atlOrigin=eyJpIjoiNTJjMGU2Mjc4NGIwNGNiNzhkZmU4ZmFjMWZjYTYyMjMiLCJwIjoiaiJ9

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

image

image

near-can-stake

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: libs/ledger-live-common/src/platform/providers/RampCatalogProvider/types.ts [hunk 1]
```diff
@@ -8,47 +10,26 @@ export type PaymentServiceProvider =
...
+>;
 
 export type RampCatalog = {
-  onRamp: RampCatalogEntry[];
-  offRamp: RampCatalogEntry[];
+  onRamp: CurrenciesPerProvider;
+  offRamp: CurrenciesPerProvider;
 };
+
+export type BuyServiceProvider =
```
Comment (cpellizzari-ledger) [inline libs/ledger-live-common/src/platform/providers/RampCatalogProvider/types.ts:23]: This shouldn't be a closed enum actually, as we want to be able to add new providers in Buy app / Buy service without changing the LL.

I believe this type is not checked at runtime anyway, but I prefer to have explicit forward compatibility with any string allowed.

File: apps/ledger-live-desktop/src/renderer/screens/asset/AssetBalanceSummaryHeader.tsx [hunk 2]
```diff
@@ -71,23 +70,25 @@ export default function AssetBalanceSummaryHeader({
...
-  const availableOnSwap = providers?.concat(storedProviders ?? []).some(({ pairs }) => {
-    return pairs && pairs.find(({ from, to }) => [from, to].includes(currency.id));
-  });
+
+  const availableOnSwap = providers
+    ?.concat(storedProviders ?? [])
+    .some(({ pairs }: { pairs: { from: string; to: string }[] }) => {
+      return (
+        pairs &&
+        pairs.find(({ from, to }: { from: string; to: string }) => [from, to].includes(currency.id))
```
Comment (mitchellv-ledger) [inline apps/ledger-live-desktop/src/renderer/screens/asset/AssetBalanceSummaryHeader.tsx:89]: negligible improvement (so not necessary) but likely performance better if just comparing directly as `from === currency.id || to === curency.id` as you're creating a new array on each iteration

File: libs/ledger-live-common/src/platform/providers/RampCatalogProvider/useRampCatalog.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    currencyId: CryptoCurrency["id"] | string,
+    mode: "onRamp" | "offRamp",
+  ) => boolean | null;
+  getSupportedCryptoCurrencyIds: (mode: "onRamp" | "offRamp") => CryptoCurrency["id"][] | null;
+};
+
+export function useRampCatalog(): UseRampCatalog {
+  const state = useRampCatalogContext();
+
+  useEffect(() => {}, [state.value]);
```
Comment (mitchellv-ledger) [inline libs/ledger-live-common/src/platform/providers/RampCatalogProvider/useRampCatalog.ts:17]: This doesn't do anything?

**Review by mitchellv-ledger — CHANGES_REQUESTED** [review]
Looks great, a couple of small things to address

File: libs/ledger-live-common/src/platform/providers/RampCatalogProvider/useRampCatalog.ts (hunk 3 above)
Comment (beths-ledger) [reply libs/ledger-live-common/src/platform/providers/RampCatalogProvider/useRampCatalog.ts:18]: aha yes a hangover from a console log, thanks for spotting!!

**Review by Josemoe — COMMENTED** [review]
Josemoe21

**Comment by gre** [comment]
as there are a screenshot conflict, please rebase and /generate-screenshots when the CI is stable on develop again

**Pull request merged**